bridges/NyaaTorrentsBridge.php

```diff
@@ -0,0 +1,127 @@
+<?php
+class NyaaTorrentsBridge extends BridgeAbstract {
+
+	const MAINTAINER = 'ORelio';
+	const NAME = 'NyaaTorrents';
+	const URI = 'https://nyaa.si/';
+	const DESCRIPTION = 'Returns the newest torrents, with optional search criteria.';
+	const PARAMETERS = array(
+		array(
+			'f' => array(
+				'name' => 'Filter',
+				'type' => 'list',
+				'values' => array(
+					'No filter' => '0',
+					'No remakes' => '1',
+					'Trusted only' => '2'
+				)
+			),
+			'c' => array(
+				'name' => 'Category',
+				'type' => 'list',
+				'values' => array(
+					'All categories' => '0_0',
+					'Anime' => '1_0',
+					'Anime - AMV' => '1_1',
+					'Anime - English' => '1_2',
+					'Anime - Non-English' => '1_3',
+					'Anime - Raw' => '1_4',
+					'Audio' => '2_0',
+					'Audio - Lossless' => '2_1',
+					'Audio - Lossy' => '2_2',
+					'Literature' => '3_0',
+					'Literature - English' => '3_1',
+					'Literature - Non-English' => '3_2',
+					'Literature - Raw' => '3_3',
+					'Live Action' => '4_0',
+					'Live Action - English' => '4_1',
+					'Live Action - Idol/PV' => '4_2',
+					'Live Action - Non-English' => '4_3',
+					'Live Action - Raw' => '4_4',
+					'Pictures' => '5_0',
+					'Pictures - Graphics' => '5_1',
+					'Pictures - Photos' => '5_2',
+					'Software' => '6_0',
+					'Software - Apps' => '6_1',
+					'Software - Games' => '6_2',
+				)
+			),
+			'q' => array(
+				'name' => 'Keyword',
+				'description' => 'Keyword(s)',
+				'type' => 'text'
+			)
+		)
+	);
+
+	public function collectData() {
+
+		// Build Search URL from user-provided parameters
+		$search_url = self::URI . '?s=id&o=desc&'
+		. http_build_query(array(
+			'f' => $this->getInput('f'),
+			'c' => $this->getInput('c'),
+			'q' => $this->getInput('q')
+		));
+
+		// Retrieve torrent listing from search results, which does not contain torrent description
+		$html = getSimpleHTMLDOM($search_url)
+		or returnServerError('Could not request Nyaa: '.$search_url);
+		$links = $html->find('a');
+		$results = array();
+		foreach ($links as $link)
+			if (strpos($link->href, '/view/') === 0 && !in_array($link->href, $results))
+				$results[] = $link->href;
+		if (empty($results))
+			returnServerError('No results from Nyaa: ' . $url, 500);
```

Comment:
Same as the previous bridge, in case of non-matching search, you should just return an empty feed